packages/openapi-generator/src/file-serializer/interface-file.ts

```diff
@@ -0,0 +1,37 @@
+import { codeBlock } from '@sap-cloud-sdk/util';
+import { OpenApiNamedSchema, OpenApiSchema } from '../openapi-types';
+import {
+  collectRefs,
+  hasNotSchema,
+  parseTypeNameFromRef,
+  parseFileNameFromRef
+} from '../model';
+import { serializeSchema } from './schema';
+import { Import, serializeImports } from './imports';
+
+/**
+ * Serialize a schema representation to a string representing the according interface file contents.
+ * @param operationInfo A named schema.
+ * @returns The serialized interface file contents.
+ */
+export function interfaceFile({ name, schema }: OpenApiNamedSchema): string {
+  const imports = serializeImports(getImports(schema));
+
+  return codeBlock`
+    ${imports}
+    export type ${name} = ${serializeSchema(schema)};
+  `;
+}
+
+function getImports(schema: OpenApiSchema): Import[] {
+  const refImports = collectRefs(schema).map(ref => ({
+    names: [parseTypeNameFromRef(ref)],
+    typeOnly: true,
+    moduleIdentifier: `./${parseFileNameFromRef(ref)}`
+  }));
+  const coreImportNames = hasNotSchema(schema) ? ['Except'] : [];
```

Comment:
The `Except` is the only core dependency so line 34 makes no sense with no names given in the case of hasNotSchema === false.

Reply:
in TODOs